packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js

```diff
@@ -0,0 +1,478 @@
+   * @returns {Promise<BaseModel>} - A promise that resolves to the created model instance.
+   * @throws {Error} - Throws an error if the data is invalid or if the creation process fails.
+   */
+  async create(item) {
+    if (!isNonEmptyObject(item)) {
+      const message = `Failed to create [${this.entityName}]: data is required`;
+      this.log.error(message);
+      throw new Error(message);
+    }
+
+    try {
+      const record = await this.entity.create(item).go();
+      return this.#createInstance(record.data);
+    } catch (error) {
+      this.log.error(`Failed to create [${this.entityName}]`, error);
+      throw error;
+    }
+  }
+
+  /**
+   * Validates and batches items for batch operations.
+   * @private
+   * @param {Array<Object>} items - Items to be validated.
+   * @returns {Object} - An object containing validated items and error items.
+   */
+  #validateItems(items) {
+    const validatedItems = [];
+    const errorItems = [];
+
+    items.forEach((item) => {
+      try {
+        const { Item } = this.entity.put(item).params();
+        validatedItems.push({ ...removeElectroProperties(Item), ...item });
+      } catch (error) {
+        if (error instanceof ElectroValidationError) {
+          errorItems.push({ item, error: new ValidationError(error) });
+        }
+      }
+    });
+
+    return { validatedItems, errorItems };
+  }
+
+  /**
+   * Creates multiple entities in the collection and directly persists them to the database in
+   * a batch write operation. Batches are written in parallel and are limited to 25 items per batch.
+   *
+   * @async
+   * @param {Array<Object>} newItems - An array of data for the entities to be created.
+   * @param {BaseModel} [parent] - Optional parent entity that these items are associated with.
+   * @return {Promise<{ createdItems: BaseModel[],
+   * errorItems: { item: Object, error: ValidationError }[] }>} - A promise that resolves to
+   * an object containing the created items and any items that failed validation.
+   * @throws {ValidationError} - Throws a validation error if any of the items has validation
+   * failures.
+   */
+  async createMany(newItems, parent = null) {
+    if (!Array.isArray(newItems) || newItems.length === 0) {
+      const message = `Failed to create many [${this.entityName}]: items must be a non-empty array`;
+      this.log.error(message);
+      throw new Error(message);
+    }
+
+    try {
+      const { validatedItems, errorItems } = this.#validateItems(newItems);
+
+      if (validatedItems.length > 0) {
+        const response = await this.entity.put(validatedItems).go();
+
+        if (Array.isArray(response?.unprocessed) && response?.unprocessed?.length > 0) {
+          this.log.error(`Failed to process all items in batch write for [${this.entityName}]: ${JSON.stringify(response.unprocessed)}`);
+        }
+      }
+
+      const createdItems = this.#createInstances(validatedItems);
```

Comment:
If a validated item fails to be added to the database using the `put` command,  it still creates an instance and returns the item as successful like this. 
I would expect the method to return the failed/unprocessed items in the `errorItems` instead

Reply:
i'd like to table this for now, it only happens if throughput capacity is exceeded.